### PR TITLE
Update to current fastest env setitem

### PR DESCRIPTION
**Pull request by mwichmann**
Updated and re-ran benchmark tests for `SubstitutionEnvironment.__setitem__`. Added test for new (Python 3.0) string.`isidentifier()` method. That's actually exactly what we want, and it times out as the fastest method when combined with a membership test if the variable is already defined.

Tweaked some comments about this performance consideration, and did other updates in `bench/`.

No doc impacts as there are no behavioral changes.

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `CHANGES.txt` (and read the `README.rst`)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: bench/env.__setitem__.py [hunk 1]
```diff
@@ -258,6 +258,18 @@ def __setitem__(self, key, value):
                     raise SCons.Errors.UserError("Illegal construction variable `%s'" % key)
             self._dict[key] = value
 
+class env_Best_has_key_function(Environment):
```
Comment (bdbaddog) [inline bench/env.__setitem__.py:261]: Is this just so we have a comparison against the current implementation?

File: bench/is_types.py [hunk 2]
```diff
@@ -51,7 +53,7 @@ def cache_type_e_is_Dict(e):
 
 def cache_type_e_is_List(e):
     t = type(e)
-    return t is list or isinstance(e, UserList)
+    return t is list or isinstance(e, UserList or isinstance(e, deque))
```
Comment (bdbaddog) [inline bench/is_types.py]: Is this right? `or` inside the isinstance()?

Comment (mwichmann) [reply bench/is_types.py]: that got clumsy. I'll fix - think should be single isinstance

File: bench/env.__setitem__.py (hunk 1 above)
Comment (mwichmann) [reply bench/env.__setitem__.py:261]: yes.

File: bench/is_types.py (hunk 2 above)
Comment (mwichmann) [reply bench/is_types.py]: yeah, it was supposed to look like the one a few lines later.  pushed update.

**Pull request merged**